virtualmachine/vsphere/vm.go

```diff
@@ -1393,3 +1428,62 @@ func CreateTemplate(vm *VM) error {
 	err = uploadTemplate(vm, dcMo, vm.datastore)
 	return err
 }
+
+// GetDatastores : Returns the datastores in a host/cluster in a cluster
+func GetTemplateList(vm *VM) ([]map[string]string, error) {
+	vmMoList, err := getVirtualMachines(vm)
+	if err != nil {
+		return nil, err
+	}
+
+	vmList := make([]map[string]string, 0)
+	for _, vmo := range vmMoList {
+		// Filter out the templates
+		if vmo.Config.Template {
+			vmList = append(vmList, map[string]string{
+				"name": vmo.Name,
+				"id":   vmo.Self.Value,
+			})
+		}
+	}
+	return vmList, nil
+}
+
+// GetVirtualMachines : Return the virual machines in a cluster
+func getVirtualMachines(vm *VM) ([]mo.VirtualMachine, error) {
```

Comment:
I had kept this private since it is returning a list of objects of type defined in govmomi. The rest of the exposed methods are returning generic types.
Let me know if this still needs to be exposed. I'll change the definition